### DISCUSSION TITLE
Cart and Checkout objects are not synced

### DISCUSSION THREAD

**Discussion by icrustandi** [Shopify/storefront-api-feedback#23] [Help] [answered]
Hi,

As we're trying to move off from using the checkout object to implement our cart on storefront API using the Cart endpoint, I noticed that Cart and Checkout objects are not synced. Steps to reproduce:
1. customer has a cart with the following items: [{variant A, qty: 1}, {variant B, qty:1}]
2. customer goes to checkout via checkoutUrl in cart object, sees that the line items correspond with the cart
3. customer goes back to the same cart object, adds another item and/or updates the qty on existing line items: [{variant A, qty: 2}, {variant B, qty:3}, {variant C, qty:1}]
4. customer goes to the same checkoutUrl and sees that the state of the checkout is stale, with the items in the checkout corresponding to number 1.

The expectation here is that the customer should see in their checkout the items in the cart on number 3. This de-sync is true when applying discount codes. If a customer applies a discount code during checkout, the cart object does not reflect the change. 

Will this be fixed?

**Comment by ChrisKG32** [comment, marked as answer]
The dev team is aware of this and working on a fix
https://github.com/Shopify/storefront-api-feedback/discussions/16#discussioncomment-1023513
